src/sentry/static/sentry/app/views/alerts/details/chart.tsx

```diff
@@ -37,6 +37,33 @@ function getNearbyIndex(data: Data, needle: number) {
   return index !== -1 ? index - 1 : data.length - 1;
 }
 
+/**
+ * We can't just pass an x value to the charts, so we calculate a y value
+ * between points using the average of the two points it's between.
+ *
+ * @param data Data array
+ * @param index The (lower) index of the two points used to calculate the average
+ */
+function getAverageBetweenPoints(data: Data, index: number) {
+  if (index >= data.length - 1) {
+    return getDataValue(data[data.length - 1]);
+  } else if (index < 0) {
+    return getDataValue(data[0]);
+  } else {
+    const pt1 = getDataValue(data[index]);
+    const pt2 = getDataValue(data[index + 1]);
+    return (pt1 + pt2) / 2;
+  }
+}
+
+function getDataValue(data: [number, {count: number}[]]) {
```

Comment:
Can we do some typescript refactoring? Currently `Data` is typed as an array of this:

```
type Data = [number, {count: number}[]][];
```

`getDataValue` accepts a single "data" of this type which can be typed as `Data[number]`, but I think it might be a bit more readable if `Data` was defined as the singular version of this:

```
type Data = [number, Array<{count: number}>];
```

and refactoring existing usages of `Data` to be an array of `Data`: (`Data[]`)